train.py

```diff
@@ -329,66 +368,143 @@ def run_epoch(epoch, model, training, metric_tracker, data_loader):
         for tracker_name in ["training", "validation", "testing"]
     )
 
+    epoch_progress = Progress(
+        "{task.description}",
+        SpinnerColumn(),
+        BarColumn(),
+        TextColumn("{task.percentage:>3.0f}%"),
+        metric_tracker_train.get_metric_text_column(),
+    )
+
+    train_progress_dict = {
+        **dict(description="[green]Training", total=len(train_set_loader)),
+        **metric_tracker_train.get_current_iteration_metric_text_column_fields(),
+    }
+
+    val_progress_dict = {
+        **dict(description="[yellow]Validation", total=len(val_set_loader)),
+        **metric_tracker_val.get_current_iteration_metric_text_column_fields(),
+    }
+
+    train_epoch_progress = epoch_progress.add_task(**train_progress_dict)
+    val_epoch_progress = epoch_progress.add_task(**val_progress_dict)
+
+    for task in epoch_progress.tasks:
+        print(task)
+
+    overall_progress = Progress()
+
+    total_iters = (args.max_epochs - 1 - start_epoch) * (
+        len(train_set_loader) + len(val_set_loader)
+    ) + len(test_set_loader) * args.test
+
+    overall_task = overall_progress.add_task(
+        "Experiment Progress",
+        completed=(start_epoch + 1) * (len(train_set_loader) + len(val_set_loader)),
+        total=total_iters,
+    )
+
+    progress_table = Table.grid()
+    progress_table.add_row(
+        Panel.fit(
+            overall_progress,
+            title="Experiment Progress",
+            border_style="green",
+            padding=(2, 2),
+        ),
+    )
+    progress_table.add_row(
+        Panel.fit(epoch_progress, title="[b]Jobs", border_style="red", padding=(1, 2))
+    )
+    progress_table.add_row(metric_tracker_train.per_epoch_table)
+    progress_table.add_row(metric_tracker_val.per_epoch_table)
+
```

Comment:
this whole section here is quite verbose and doesn't have much to do with training. can we move it into a utils function?

Reply:
I was considering that too. I wanted to see what you thought before I added more abstraction.

Reply:
I could easily have a whole class taking care of the whole interface thing. That would make everything a lot neater.